lib/db.js

```diff
@@ -49,69 +86,152 @@ module.exports = function(dyno, kinesis, config) {
-          status: {EQ: 'leased'},
-          instance: {EQ: config.instanceId}
+    dyno.updateItem({
+      Key: {type:'shard', id: s.id},
+      AttributeUpdates: {
+        counter: {
+          Action: 'ADD',
+          Value: 1
+        },
+        expiresAt: {
+          Action: 'PUT',
+          Value: +new Date() + config._leaseTimeout
+        },
+        updated: {
+          Action: 'PUT',
+          Value: +new Date()
         }
-      }, cb);
+      },
+      ConditionalOperator: 'AND',
+      Expected: {
+        expiresAt: {
+          ComparisonOperator: 'GE',
+          AttributeValueList: [+new Date()]
+        },
+        status: {
+          ComparisonOperator: 'EQ',
+          AttributeValueList: ['leased']
+        },
+        instance: {
+          ComparisonOperator: 'EQ',
+          AttributeValueList: [config.instanceId]
+        }
+      },
+      ReturnValues: 'ALL_NEW'
+    }, function (err, response) {
+      var updated = err ? response : response.Attributes;
+      return cb(err, updated);
+    });
   };
 
   db.shardComplete = function(s, cb) {
-    dyno.updateItem(
-      {type:'shard', id: s.id},
-      {put: {status: 'complete'}},
-      {
-        conditionalOperator: 'AND',
-        expected: {
-          expiresAt: {GE: +new Date()},
-          status: {EQ: 'leased'},
-          instance: {EQ: config.instanceId}
+    dyno.updateItem({
+      Key: {type:'shard', id: s.id},
+      AttributeUpdates: {
+        status: {
+          Action: 'PUT',
+          Value: 'complete'
         }
-      }, cb);
+      },
+      ConditionalOperator: 'AND',
+      Expected: {
+        expiresAt: {
+          ComparisonOperator: 'GE',
+          AttributeValueList: [+new Date()]
+        },
+        status: {
+          ComparisonOperator: 'EQ',
+          AttributeValueList: ['leased']
+        },
+        instance: {
+          ComparisonOperator: 'EQ',
+          AttributeValueList: [config.instanceId]
+        }
+      },
+      ReturnValues: 'ALL_NEW'
+    }, function (err, response) {
+      var updated = err ? response : response.Attributes;
+      return cb(err, updated);
+    });
   };
 
   db.updateInstance = function(instance, cb) {
-    dyno.updateItem(
-      {type:'instance', id: config.instanceId},
-      {add: {counter: 1}, put: {expiresAt: +new Date() + config._leaseTimeout}},
-      cb
-    );
+    dyno.updateItem({
+      Key: {type:'instance', id: config.instanceId},
+      AttributeUpdates: {
+        counter: {
+          Action: 'ADD',
+          Value: 1
+        },
+        expiresAt: {
+          Action: 'PUT',
+          Value: +new Date() + config._leaseTimeout
+        }
+      },
+      ReturnValues: 'ALL_NEW'
+    }, function (err, response) {
+      var updated = err ? response : response.Attributes;
+      return cb(err, updated);
+    });
   };
 
   db.cleanupInstances = function(cb) {
-    dyno.query({type:{EQ:'instance'}}, {pages:0}, function(err, instances){
+    dyno.query({
```

Comment:
wont be a problem at any scale we have run kine so far, if there are enough instances we'd want to have paging here